FraudProofs.md

```diff
@@ -0,0 +1,154 @@
+these oracles attesting to these outcomes.
+This signature point is used to validate that `aggregate_oracle_attestation*G = S` which proves that there must exist an attestation
+from each of the oracles in `oracle_announcements` of their corresponding outcome in `oracle_outcomes`.
+
+Finally, this proof references a specific oracle by `oracle_index` which is the index in `oracle_announcements` and `oracle_outcomes`
+corresponding to the referenced oracle's announcement and outcome to which they attested. 
+
+## Claim of False Oracle Attestation
+
+The most straightforward kind of oracle fraud occurs when an oracle attests to a false event outcome,
+meaning that the attested-to outcome is not what truly occurred in the real world.
+
+Unfortunately, there is no general and automatic way to validate a claim of this kind of fraud as there
+is often times some amount of subjectivity involved or at the very least validation must compare what
+the oracle said to the real world, a process which usually requires manual human labor.
+
+As such, this specification only provides a standard format for making such claims which is generally
+intended for use with other external infrastructure.
+That said, it is possible for DLC client software to accept the manual importing of these claims after
+which user's can be taken through manual validation step-by-step.
+Such a process can serve as a manual means of black-listing oracle keys.
+
+#### Version 0 `false_oracle_attestation_claim`
+
+1. type: ??? (`false_oracle_attestation_claim_v0`)
+2. data:
+   * [`oracle_attestation_proof`:`oracle_attestation_proof`]
+   * [`string`:`fraud_claim`]
+
+This message consists of an `oracle_attestation_proof` showing that a specific oracle attested to
+a specific message and a `fraud_claim` explaining why this attestation is fraudulent.
+
+## Proof of Invalid Oracle Attestation
+
+The next simplest form of fraud is if an oracle attests using some outcome outside of its announced
+and committed to outcome set from its `oracle_announcement`.
+
+Unlike false attestations, this kind of fraud can be automatically validated by client software.
+
+#### Version 0 `invalid_oracle_attestation_proof`
+
+1. type: ??? (`invalid_oracle_attestation_proof_v0`)
+2. data:
+   * [`oracle_attestation_proof`: `oracle_attestation_proof`]
+
+This type simply consists of an `oracle_attestation_proof` proving that a specific oracle attested to
+a specific `oracle_outcome` where this outcome is not allowed by that oracle's `oracle_announcement`.
+
+An important thing to keep in mind is that in the case of invalid attestations, there are no concerns related
+to recovering the `oracle_outcome` from on-chain information because there are no CETs or adaptor
+signatures built for such an outcome.
+In other words, this kind of fraud is only detectable and only important if the `oracle_outcome` is known
```

Comment:
Shouldn't this say if the oracle_outcome is valid but the corresponding `oracle_announcement` is unknown ?

Reply:
No. If the announcement is not known then there is no feasible way to prove fraud or attestation of any kind because you don't even know what keys are let alone messages, not really sure what it would mean for an `oracle_announcement` to be unknown and for anything in this document to matter.

What is being said here is that if you know the `oracle_announcement` and `oracle_outcome` and that outcome is not in the announcement then you can prove fraud, but if you don't know the `oracle_outcome` then you can't know that a given message is an attestation of anything because without knowledge of the outcome it is just a random 32 bytes.